config/co2eq_parameters.js

```diff
@@ -1,18 +1,51 @@
 var exports = module.exports = {};
 
-var defaultCo2eqFootprint = {
-    'biomass': 230,
-    'coal': 820,
-    'gas': 490,
-    'hydro': 24,
-    'nuclear': 12,
-    'oil': 650,
-    'solar': 45,
-    'wind': 12,
-    'geothermal': 24,
-    'unknown': 700, // assume conventional
-    'other': 700 // same as 'unknown'. Here for backward compatibility
-}; // in gCo2eq/kWh
+var defaultCo2eqFootprint = { // in gCo2eq/kWh
+    'biomass': {
+      'co2eq': 230,
+      'source': defaultCo2eqSource
+    },
+    'coal': {
+      'co2eq': 820,
+      'source': defaultCo2eqSource
+    },
+    'gas': {
+      'co2eq': 490,
+      'source': defaultCo2eqSource
+    },
+    'hydro': {
+      'co2eq': 24,
+      'source': defaultCo2eqSource
+    },
+    'nuclear': {
+      'co2eq': 12,
+      'source': defaultCo2eqSource
+    },
+    'oil': {
+      'co2eq': 650,
+      'source': 'UK POST 2014' // UK Parliamentary Office of Science and Technology (2006) "Carbon footprint of electricity generation"
+    },
+    'solar': {
+      'co2eq': 45,
+      'source': defaultCo2eqSource
+    },
+    'wind': {
+      'co2eq': 12,
+      'source': defaultCo2eqSource
+    },
+    'geothermal': {
+      'co2eq': 38,
+      'source': defaultCo2eqSource
+    },
+    'unknown': {
+      'co2eq': 700, // assume conventional
+      'source': defaultCo2eqSource
+    },
+    'other': {
+      'co2eq': 700, // // same as 'unknown'. Here for backward compatibility
+      'source': defaultCo2eqSource
+    }
+};
 var defaultCo2eqSource = 'IPCC 2014';
```

Comment:
You need to move this declaration above L3, because else you're using the variable `defaultCo2eqSource` which does not exist